# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/GCMarker.h
# Commit: d33708e4f8ec
# Full Hash: d33708e4f8ecd98a449df50766a2ea59045b317f
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-10-29 09:50:10
# Regressor Bug: 1797755
# File Overlap Count: 1
# Description:
#   Bug 1797755 - Part 3: Remove the unused MarkStackIter class r=sfink
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D160526
# ==============================================================================

diff -r 967fd2ce6f67 -r d33708e4f8ec js/src/gc/GCMarker.h
--- a/js/src/gc/GCMarker.h	Fri Oct 28 15:17:42 2022 +0000
+++ b/js/src/gc/GCMarker.h	Fri Oct 28 15:17:43 2022 +0000
@@ -204,28 +204,6 @@
 #ifdef DEBUG
   mutable size_t iteratorCount_;
 #endif
-
-  friend class MarkStackIter;
-};
-
-class MarkStackIter {
-  MarkStack& stack_;
-  size_t pos_;
-
- public:
-  explicit MarkStackIter(MarkStack& stack);
-  ~MarkStackIter();
-
-  bool done() const;
-  MarkStack::Tag peekTag() const;
-  MarkStack::TaggedPtr peekPtr() const;
-  MarkStack::SlotsOrElementsRange peekSlotsOrElementsRange() const;
-  void next();
-  void nextPtr();
-  void nextArray();
-
- private:
-  size_t position() const;
 };
 
 } /* namespace gc */